### PR TITLE
Introducing Intl component to translate country names instead of Translatable

### PR DESCRIPTION
**Pull request by pamil**
PR based on #2720 (thanks @gperdomor!).

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes (there is no `USA` country no more, use `United States`)
| Deprecations? | no
| Fixed tickets | #2710, #2720 
| License       | MIT

### CODE REVIEW

File: features/frontend/checkout_shipping_i18n.feature [hunk 1]
```diff
@@ -5,7 +5,7 @@ Feature: Checkout shipping in preferred language
     I want to be able to see shipping methods in my preferred language
 
     Background:
-        Given there are following taxonomies defined:
+        Given there are following taxonomies defined:C
```
Comment (michalmarcinkowski) [inline features/frontend/checkout_shipping_i18n.feature]: What is the 'C' letter? :smile:

File: features/user/account/account_orders.feature [hunk 2]
```diff
@@ -30,18 +30,18 @@ Feature: User account orders page
...
+        | linustorvalds@linux.com | DHL                      | Linus Torvalds, Väätäjänniementie 59, 00440, Helsinki, Finland              |
+        | linustorvalds@linux.com | DHL                      | Linus Torvalds, Väätäjänniementie 59, 00440, Helsinki, Finland              |
+        | sylius@example.com      | UPS                      | Théophile Morel, 17 avenue Jean Portalis, 33000, Bordeaux, France           |
+        | sylius@example.com      | UPS                      | Théophile Morel, 17 avenue Jean Portalis, 33000, Bordeaux, France           |
+        | linustorvalds@linux.com | DHL                      | Linus Torvalds, Väätäjänniementie 59, 00440, Helsinki, Finland              |
+        | sylius@example.com      | UPS                      | Théophile Morel, 17 avenue Jean Portalis, 33000, Bordeaux, France           |
+        | sylius@example.com      | UPS                      | Théophile Morel, 17 avenue Jean Portalis, 33000, Bordeaux, France           |
+        | ianmurdock@debian.org   | FedEx                    | Ian Murdock, 3569 New York Avenue, CA 92801, San Francisco, United States   |
```
Comment (michalmarcinkowski) [inline features/user/account/account_orders.feature]: There should be only one space between 'United States' and '|' at the end

File: src/Sylius/Bundle/AddressingBundle/Behat/AddressingContext.php [hunk 3]
```diff
@@ -135,22 +137,4 @@ public function thereIsProvince($name)
...
-            $country = $countryTranslation->getTranslatable();
-            $country
-                ->setCurrentLocale($data['locale'])
-                ->setName($data['name']);
-        }
-
-        $manager->flush();
-    }
-}
+}
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AddressingBundle/Behat/AddressingContext.php]: There should be a new line at the end of file

File: src/Sylius/Bundle/AddressingBundle/Form/Type/CountryType.php [hunk 4]
```diff
@@ -28,12 +29,8 @@ class CountryType extends AbstractResourceType
...
     {
         $builder
-            ->add('translations', 'a2lix_translationsForms', array(
-                'form_type' => 'sylius_country_translation',
-                'label'    => 'sylius.form.country.name',
-            ))
-            ->add('isoName', 'text', array(
-                'label' => 'sylius.form.country.iso_name',
+            ->add('isoName', 'country', array(
+                'label' => 'sylius.form.country.name',
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AddressingBundle/Form/Type/CountryType.php]: ``` sylius.form.country.name ``` -> ``` sylius.form.country.iso_name ``` ?

File: src/Sylius/Bundle/AddressingBundle/Resources/translations/messages.en.yml [hunk 5]
```diff
@@ -11,7 +11,6 @@ sylius:
             street: Street
             province: Province
         country:
-            iso_name: ISO name
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AddressingBundle/Resources/translations/messages.en.yml]: Why?

File: src/Sylius/Component/Addressing/Model/Country.php [hunk 6]
```diff
@@ -29,14 +30,12 @@ class Country extends AbstractTranslatable implements CountryInterface
...
      */
     protected $id;
-
     /**
      * Country name in ISO format.
      *
      * @var string
      */
     protected $isoName;
-
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Addressing/Model/Country.php]: Why?

File: src/Sylius/Component/Addressing/spec/Model/CountrySpec.php [hunk 7]
```diff
@@ -64,6 +67,7 @@ function it_has_no_iso_name_by_default()
         $this->getIsoName()->shouldReturn(null);
     }
 
+
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Addressing/spec/Model/CountrySpec.php]: Why?

File: src/Sylius/Component/Addressing/spec/Model/CountrySpec.php [hunk 8]
```diff
@@ -96,7 +100,6 @@ function it_removes_province(ProvinceInterface $province)
     {
         $this->addProvince($province);
         $this->hasProvince($province)->shouldReturn(true);
-
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Addressing/spec/Model/CountrySpec.php]: I think it was more readable with this blank line

File: features/frontend/checkout_shipping_i18n.feature (hunk 1 above)
Comment (pamil) [inline features/frontend/checkout_shipping_i18n.feature]: Hidden Behat feature, don't know why it works :)

File: src/Sylius/Bundle/AddressingBundle/Form/Type/CountryType.php (hunk 4 above)
Comment (pamil) [inline src/Sylius/Bundle/AddressingBundle/Form/Type/CountryType.php]: It shows localized country name, not an iso name. It uses iso name as values in `option`.

File: src/Sylius/Bundle/AddressingBundle/Resources/translations/messages.en.yml (hunk 5 above)
Comment (pamil) [inline src/Sylius/Bundle/AddressingBundle/Resources/translations/messages.en.yml]: It's not used.

File: src/Sylius/Bundle/AddressingBundle/Form/Type/CountryType.php (hunk 4 above)
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AddressingBundle/Form/Type/CountryType.php]: :+1:

**Comment by Symfomany** [comment]
:+1: @pamil  It will bemore easily i think for everybody!

**Comment by michalmarcinkowski** [comment]
Migration script for the database is missing.

You should also write upgrade instructions in UPGRADE.md file and add this change to the changelog file (https://github.com/Sylius/Sylius/pull/2990).

Thanks! :+1:

**Comment by Symfomany** [comment]
@michalmarcinkowski Use Doctrine Migrations no?

**Comment by pamil** [comment]
@michalmarcinkowski [all done :)](https://github.com/michalmarcinkowski/Sylius/pull/4)

**Comment by pamil** [comment]
Closed in favour of #3074.

**Pull request closed without merging**